Bump minimum Ruby version to 2.4

PanisSupraOmnia opened the pull request:
This PR is for when we're ready to bump the minimum Ruby version to 2.4. This shouldn't be _all_ that soon, but Ruby 2.3 reaches EOL next March, so it's not too far away.

z64 commented:
Hi @PanisSupraOmnia , it's now March, and EOL is the 31st.

I see you pushed a couple weeks ago; If this PR is up to date, let me know and we can move this forward. Thanks!

z64 commented:
Does our Dependencies section need to be updated with this PR?

https://github.com/meew0/discordrb/blob/6f9dd82e57125092ba6c11f8c0e909193fa86b61/README.md#dependencies

PanisSupraOmnia commented:
Yes, I went ahead and updated it, and took the liberty of simplifying that part of the instructions for Windows given that the change in Devkit setup occurred in 2.4. Also, added 2.6 to the Travis test matrix/set the docs deploy to use it.

I realized that this PR's purpose is really to bump the minimum Ruby version, and that the RuboCop changes are just a part of that; I've renamed the PR accordingly.

Lastly, as can be seen by the commits, this PR is based on #587, so that PR should either be merged before this one, or just closed.

PanisSupraOmnia commented on README.md:
```diff
@@ -23,8 +23,8 @@ See also: [Documentation](https://www.rubydoc.info/gems/discordrb), [Tutorials](
 
-* Ruby >= 2.3
-* An installed build system for native extensions (on Windows for Ruby < 2.4, try the [DevKit](https://rubyinstaller.org/downloads/); installation instructions [here](https://github.com/oneclick/rubyinstaller/wiki/Development-Kit#quick-start) - you only need to do the quick start)
+* Ruby >= 2.4
```
Sure, that makes sense, since for now all anyone wanting to make it work with 2.3 needs to do is revert the changes made in this PR; this way we're just declaring that going forward we'll be writing 2.4-compatible code only.
